src/docker.ts

```diff
@@ -57,8 +57,28 @@ export async function push(imageName: string, auth: string): Promise<void> {
 }
 
 export function tag(existingImageName: string, newImageName: string): void {
-  const tagOutput = spawnSync('docker', ['tag', existingImageName, newImageName], { encoding: 'utf-8', stdio: 'inherit' })
-  if (tagOutput.status !== 0) {
-    throw new Error(tagOutput.error?.message ?? 'Error tagging image')
+  spawnDockerSync(['tag', existingImageName, newImageName], { stdio: 'inherit' })
+}
+
+function spawnDockerSync(args: string[], options?: { stdio?: StdioOptions }): SpawnSyncReturns<string> {
+  const opts: SpawnSyncOptionsWithStringEncoding = { encoding: 'utf-8', ...options }
+  const result = spawnSync('docker', args, opts)
+  const exitCode = result.status
+  if (
+    (exitCode !== null && exitCode !== 0) ||
+    result.error !== undefined
+  ) {
+    if (result.stderr?.includes('No such image')) {
+      throw new Error('Docker failed to find image. Make sure you have built the image and it\'s listed in the `docker images` command output before running this command.')
```

Comment:
how is this different from the `docker ps` command we ran when debugging?

Reply:
`docker ps` lists the running docker containers. `docker images` lists out the available images on your machine - like the ones you've pulled from a registry or that you've built. A user could run into this by running `jamsocket push my-service some-missing-image`. They need to be sure they've built the image already with something like `docker build -t my-image .`. Then they can run `jamsocket push my-service my-image`.